tests/UI/pages/BO/modules/productComments/index.js

```diff
@@ -0,0 +1,136 @@
+    if (await this.elementVisible(page, '.list-empty', 3)) {
+      return 0;
+    }
+
+    const selector = this.reviewsTableRows(table);
+    return page.$$eval(selector, divs => divs.length);
+  }
+
+  /**
+   *
+   * @param page {Page} Browser tab
+   * @param table {String} The review table (2 options available: 'waiting-approval', 'reported')
+   * @param row {Number} The review row
+   * @returns {Promise<void>}
+   */
+  async openProductReviewDropdown(page, table, row = 1) {
+    await this.waitForVisibleSelector(page, this.toggleDropdownButton(table, row));
+    await page.click(this.toggleDropdownButton(table, row));
+  }
+
+  /**
+   *
+   * @param page {Page} The browser tab
+   * @param table {String} The review table (3 options available: 'waiting-approval', 'reported', 'approved')
+   * @param row {Number} The review row (default is set to 1)
+   * @returns {Promise
+   * <{date: string,
+   * product: string,
+   * author: string,
+   * rating: string,
+   * id: string,
+   * title: string,
+   * content: string}>
+   * }
+   */
+  async getReviewDataFromTable(page, table, row = 1) {
+    return {
+      id: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'id')),
+      title: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'title')),
+      content: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'content')),
+      rating: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'rating')),
+      author: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'author')),
+      product: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'product-name')),
+      date: await this.getTextContent(page, this.reviewsTableColumn(table, row, 'date')),
+    };
+  }
+
+  /**
+  *
+  * @param page {Page} Browser tab
+  * @param table {String} The reviews table (the table is set by default)
+  * @param row {Number} The review row
+  * @returns {Promise<void>}
+  */
+  async approveReview(page, table = 'waiting-approval', row = 1) {
+    await page.click(this.approveWaitingReviewButton(table, row));
+  }
+
+  /**
+  *
+  * @param page {Page} Browser tab
+  * @param table {String} The reviews table (available options: 'waiting-approval', 'reported', 'deleted'
+  * @param row {Number} The review row
+  * @returns {Promise<void>}
+  */
+  async deleteReview(page, table, row = 1) {
+    if (table === 'waiting-approval') {
+      await this.openProductReviewDropdown(page, table, row);
+      await page.click(this.deleteReviewButton(table, row));
+      await this.waitForVisibleSelector(page, this.confirmReviewDeletionButton);
+      await page.click(this.confirmReviewDeletionButton);
+    } else {
+      await page.click(this.deleteReviewButton(table, row));
+      await this.waitForVisibleSelector(page, this.confirmReviewDeletionButton);
+      await page.click(this.confirmReviewDeletionButton);
+    }
+  }
```

Comment:
```suggestion
  async deleteReview(page, table, row = 1) {
    await page.click(this.deleteReviewButton(table, row));
    await this.waitForVisibleSelector(page, this.confirmReviewDeletionButton);
    await this.clickAndWaitForNavigation(page, this.confirmReviewDeletionButton);
  }
  
  async deleteWaitingApprouvalReview(page, row = 1) {
    // Need to open dropdown before delete
    await this.openProductReviewDropdown(page, 'waiting-approval', row);
    await this.deleteReview(page, 'waiting-approval', row);
  }
  
  async deleteReportedReview(page, row = 1) {
    await this.deleteReview(page, 'reported', row);
  }
  
```

You can simplify your functions like above. And use the specific table function for delete and not the generic one.
(This is just an example, maybe there is more functions to add)

Reply:
And Maybe use a navigation to wait for the action (line 114)